arelle/plugin/validate/NL/rules/fr_nl.py

```diff
@@ -131,13 +150,16 @@ def rule_fr_nl_1_01(
     modelXbrl = val.modelXbrl
     for doc in modelXbrl.urlDocs.values():
         if doc.type == ModelDocument.Type.INSTANCE:
-            with modelXbrl.fileSource.file(doc.filepath)[0] as file:
-                if file.read(1) == '\ufeff':
-                    yield Validation.error(
-                        codes='NL.FR-NL-1.01',
-                        msg=_('A BOM (byte order mark) character MUST NOT be used in an XBRL instance document.'),
-                        fileName=doc.basename,
-                    )
+            with modelXbrl.fileSource.file(doc.filepath, binary=True)[0] as file:
+                firstLine = file.readline()
+                for bom in BOM_BYTES:
```

Comment:
Some of these are subsets and will raise multiple errors. Such as `BOM_UTF16_LE` and `BOM_UTF32_LE`. You could sort by length and break on the first match.